easybuild/tools/options.py

```diff
@@ -724,6 +730,31 @@ def postprocess(self):
             self.show_config()
             cleanup_and_exit(self.tmpdir)
 
+    def _postprocess_optarch(self):
+        """Postprocess --optarch option."""
+        optarch_parts = self.options.optarch.split(OPTARCH_SEP)
+        n_compilers = len(optarch_parts)
+        
+        # if the format is not adequate we refuse continuing
+        if n_compilers > 1 and n_compilers != self.options.optarch.count(OPTARCH_MAP_CHAR):
+            raise EasyBuildError("The optarch option has an incorrect syntax: %s" % self.options.optarch)
+        else:
+            logger = fancylogger.getLogger()
```

Comment:
In a method (where `self` is an argument), you should use `self.log`.

In a function (no `self` in scope, like in `process_software_build_specs`), you may need to create a separate logger.

In this case, there's actually a global `_log` available, we should be using that consistently where a `self.log` is not available, e.g. in `process_software_build_specs`